### PR TITLE
FIX: Amazon onebox not working.

### PR DESCRIPTION
**Pull request by tgxworld**
Fix: https://meta.discourse.org/t/amazon-onebox-issue/29066

screenshot from 2015-06-22 19 10 32

### CODE REVIEW

File: lib/onebox/engine/amazon_onebox.rb [hunk 1]
```diff
@@ -30,7 +32,11 @@ def match
       def image
         case
         when raw.css("#main-image").any?
-          raw.css("#main-image").first["src"]
+          ::JSON.parse(
+            raw.css("#main-image").first
+              .attributes["data-a-dynamic-image"]
+              .value
+          ).keys.first
```
Comment (tgxworld) [inline lib/onebox/engine/amazon_onebox.rb]:
```html
<script type="text/javascript">
if(typeof encodedImgSrc != 'undefined' && encodedImgSrc != "" && document.getElementById("main-image")!=null){document.getElementById("main-image").src=encodedImgSrc;};setCSMReq('af');setCSMReq('af');
addlongPoleTag('af','awudplphtml');
</script>
```

```html
<img alt="" src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7" onload="" class="a-dynamic-image" id="main-image" data-a-dynamic-image='{"http://ecx.images-amazon.com/images/I/51opYcR6kVL._SY400_.jpg":[200,166]}'>
```

There has been some changes with Amazon that sets the `src` of the image using Javascript.

**Comment by SamSaffron** [comment]
looks like the spec is failing, the fix is tricky, wonder how long till amazon break it again :)

File: lib/onebox/engine/amazon_onebox.rb [hunk 2]
```diff
@@ -38,15 +44,15 @@ def image
 
       def data
         result = { link: link,
-                   title: raw.css("h1").inner_text,
+                   title: raw.css("title").inner_text,
```
Comment (tgxworld) [inline lib/onebox/engine/amazon_onebox.rb]: Updated this because we can consistently get the `title` even if all else fails. :smile_cat:

**Comment by tgxworld** [comment]
> looks like the spec is failing, the fix is tricky, wonder how long till amazon break it again :)

Ah I didn't run all the specs on my local machine :stuck_out_tongue:

**Pull request merged**